cmd/vicadmin/vicadm.go

```diff
@@ -300,56 +325,76 @@ type datastoreReader struct {
 	path string
 }
 
-// find datastore logs for the appliance itself and all containers
-func findDatastoreLogs(c *session.Session) ([]entryReader, error) {
+// listVMPaths returns an array of datastore paths for VMs assocaited with the
+// VCH - this includes containerVMs and the appliance
+func listVMPaths(ctx context.Context, s *session.Session) ([]url.URL, error) {
 	defer trace.End(trace.Begin(""))
 
-	ds := object.NewDatastore(c.Vim25(), datastore)
-	ds.InventoryPath = datastoreInventoryPath
-
-	var readers []entryReader
-	ctx := context.Background()
-	b, err := ds.Browser(ctx)
-	if err != nil {
-		return nil, err
-	}
+	var err error
+	var children []*vm.VirtualMachine
 
-	spec := types.HostDatastoreBrowserSearchSpec{
-		MatchPattern: []string{"vmware.log", "*.debug"},
+	if len(vchConfig.ComputeResources) == 0 {
+		return nil, errors.New("compute resources is empty")
 	}
 
-	task, err := b.SearchDatastoreSubFolders(ctx, ds.Path(config.vmPath), &spec)
-	if err != nil {
+	ref := vchConfig.ComputeResources[0]
+	rp := compute.NewResourcePool(ctx, s, ref)
+	if children, err = rp.GetChildrenVMs(ctx, s); err != nil {
 		return nil, err
 	}
 
-	info, err := task.WaitForResult(ctx, nil)
-	if err != nil {
-		return nil, err
-	}
+	log.Infof("Found %d candidate VMs in resource pool %s for log collection", len(children), ref.String())
 
-	if info == nil {
-		return nil, errors.New("empty search results")
-	}
+	directories := []url.URL{}
+	for _, child := range children {
+		path, err := child.DSPath(ctx)
+		if err != nil {
+			log.Errorf("Unable to get datastore path for child VM %s: %s", child.Reference(), err)
+			// we need to get as many logs as possible
+			continue
+		}
 
-	if info.Error != nil {
-		return nil, errors.New(info.Error.LocalizedMessage)
+		log.Debugf("Adding VM for log collection: %s", path.String())
```

Comment:
By a, I mean in the diff above I see:
`+		Debugf("Adding VM for log collection: %s", path.String())`

but when I paste from the diff:
`+		log.Debugf("Adding VM for log collection: %s", path.String())`

I have tried deleting and recreating the line in case of non-printable characters, but it does compile and that had no effect.